### DISCUSSION TITLE
Implementing command to change password

### DISCUSSION THREAD

**Discussion by sharvari1998** [daniele77/cli#237] [Q&A] [answered]
I'm developing a CLI which will replace bash as the user's shell on Linux. For this I need to read the password from the user and confirmation of the password, without displaying it on the terminal. I need to either directly execute the passwd command and let it take these inputs, or I need to read these from my CLI and pass it on to the passwd command.
 
I tried using std::getline, std::cin.getline and also executing passwd command using system() and popen() but in all instances it is not able to read all the characters entered by the user.
 
Is there a way to achieve this using this CLI framework? Any suggestions or workarounds?

**Comment by daniele77** [comment]
I guess discussion #204 should answer your question.

**Reply by sharvari1998** [reply]
Hello , Thank you for your response,
Yes, I did go through that and I was using CliLocalTerminalSession , but was having issue with executing passwd module command using system command , I assume any command which  takes input will have this issue?

**Reply by daniele77** [reply, marked as answer]
I've just committed a fix that should enable user keyboard handling on Linux platform as well (commit 17dafd8).
It's the issue #239.
Please, see if this solves your problem.

**Reply by sharvari1998** [reply]
Hello ,
Yes this resolved our issue, Thank you for the help and great CLI framework!

**Reply by daniele77** [reply]
Very glad this solved your use case.
If you like this library, please consider supporting our work with a donation using the heart icon (sponsor) at the top of the main project page.
